metaflow/metaflow_runner.py

```diff
@@ -0,0 +1,83 @@
+import os
+import sys
+import asyncio
+import tempfile
+import aiofiles
+from typing import Dict
+from metaflow import Run
+from metaflow.cli import start
+from metaflow.click_api import MetaflowAPI
+from metaflow.subprocess_manager import SubprocessManager, CommandManager
+
+
+async def read_from_file_when_ready(file_path):
+    async with aiofiles.open(file_path, "r") as file_pointer:
+        content = await file_pointer.read()
+        while not content:
+            await asyncio.sleep(0.1)
+            content = await file_pointer.read()
+        return content
+
+
+class ExecutingRun(object):
+    def __init__(self, command_obj: CommandManager, run_obj: Run) -> None:
+        self.command_obj = command_obj
+        self.run_obj = run_obj
+
+    def __getattr__(self, name: str):
+        if hasattr(self.run_obj, name):
```

Comment:
need to think a bit more about this. We may want a more explicit control to make sure we expose what we want. It's also a bit harder to document this way.

Reply:
can discuss more in the meeting...